### PR TITLE
added pds4 gen for hyb2

### PR DESCRIPTION
**Pull request by Kelvinrr**
## Description

This gets basic PDS4 export to hayabusa2 for the ONC instrument. It still needs ISIS table info and history info which apparently is required for archiving reasons? Not sure what is the best way other than to manually add them. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [ ] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/hayabusa2/apps/hyb2pds4gen/hyb1pds4gen.xml [hunk 1]
```diff
@@ -0,0 +1,112 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<application name="hyb1pds4gen" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://isis.astrogeology.usgs.gov/Schemas/Application/application.xsd">
```
Comment (kaitlyndlee) [inline isis/src/hayabusa2/apps/hyb2pds4gen/hyb1pds4gen.xml]: Just need to go through and update references to hayabusa1 to hayabusa2

File: isis/src/hayabusa2/apps/hyb2pds4gen/hyb1pds4gen.xml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  <brief>
+    Convert Hayabusa1 image from from ISIS3 cube to PDS4 format
+  </brief>
+
+  <description>
+   This programs exports an ISIS3 Haybusa cube to a PDS4 product. 
+   If available, Instrument and  Mapping information will be written to the detached 
+   output PDS4 formatted xml label file.
+    <p>
+       Note that not all require keyword values in the generated PDS4 labels can be determined automatically
```
Comment (kaitlyndlee) [inline isis/src/hayabusa2/apps/hyb2pds4gen/hyb1pds4gen.xml]: Typo:  require -> required

File: isis/src/hayabusa2/apps/hyb2pds4gen/hyb1pds4gen.xml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+       Some of these TBDs can only be replaced with one of several specific enumerated values to meet compliancy. 
+       These possible values can be found in the 
+       <a href="https://pds.jpl.nasa.gov/pds4/doc/im/current/index_1900.html">PDS4 Information Model Specification</a> 
+       in the Value/Class column of the table for the associated tag or in the index at the end of the document. 
+       Searching this page for the name of the field you need to determine a value for is recommended. 
+    </p>
+    
+    <p>
+      For example: For a cube with map-projected data, ISIS3 will populate planar_coordinate 
+      encoding_method with a value on its own, but if it did not and instead had "TBD", we would find at
```
Comment (kaitlyndlee) [inline isis/src/hayabusa2/apps/hyb2pds4gen/hyb1pds4gen.xml]: is it supposed to be planar_coordinate_encoding_method?

File: isis/src/hayabusa2/apps/hyb2pds4gen/main.cpp [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+void IsisMain() {
+  UserInterface &ui = Application::GetUserInterface();
+
+  // Check if input file is indeed, a cube
+  if (ui.GetFileName("FROM").right(3) != "cub") {
+    QString msg = "Input file [" + ui.GetFileName("FROM") +
+                  "] does not appear to be a cube";
+    throw  IException(IException::User, msg, _FILEINFO_);
+  }
```
Comment (kaitlyndlee) [inline isis/src/hayabusa2/apps/hyb2pds4gen/main.cpp]: Do we need this check if we have a filter on the input? I know that IsisAml verifies the value of the input, like when there is a list of options for that parameter. Just want to make sure we don't have any unreachable code.

File: isis/src/hayabusa2/apps/hyb2pds4gen/hyb1pds4gen.xml (hunk 2 above)
Comment (Kelvinrr) [reply isis/src/hayabusa2/apps/hyb2pds4gen/hyb1pds4gen.xml]: I removed all the docs with the plan of re-writing them when I more of it done later.

File: isis/src/hayabusa2/apps/hyb2pds4gen/main.cpp (hunk 4 above)
Comment (Kelvinrr) [reply isis/src/hayabusa2/apps/hyb2pds4gen/main.cpp]: Carryover from hyb1pds4gen, but yeah I think this is an unnecessary check.

File: isis/src/hayabusa2/apps/hyb2pds4gen/main.cpp [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    const QString delimeter = ",";
+
+    // Output the result
+    fstream outFile;//QFile???
+    QString outputImage = ui.GetAsString("TO");
+    FileName imgOutput(outputImage);
+    FileName csvOutput = imgOutput.removeExtension().setExtension("csv");
+
+    PvlGroup instGroup = inputCubeLabel.findObject("IsisCube").findGroup("Instrument");
+    if (instGroup["InstrumentId"][0].compare("amica", Qt::CaseInsensitive) == 0) {
```
Comment (kaitlyndlee) [inline isis/src/hayabusa2/apps/hyb2pds4gen/main.cpp]: I know this is carried over from hybpds4gen, but should we update any references to amica now?

**Review by scsides — APPROVED** [review]
Good first round. Approving and merging so I can pick it up.

**Pull request merged**